mlflow_elasticsearchstore/models.py

```diff
@@ -0,0 +1,123 @@
+from mlflow.entities import (
+    Experiment, RunTag, Metric, Param, RunData, RunInfo,
+    SourceType, RunStatus, Run, ViewType, ExperimentTag)
+from mlflow.entities.lifecycle_stage import LifecycleStage
+from elasticsearch_dsl import Document, Text, Keyword, InnerDoc, Nested, Float, Integer
+
+
+class ElasticExperimentTag(InnerDoc):
+    key = Keyword()
+    value = Text()
+
+    def to_mlflow_entity(self):
+        return ExperimentTag(key=self.key,
+                             value=self.value)
+
+
+class ElasticExperiment(Document):
+
+    experiment_id = Keyword()
+    name = Keyword()
+    artifact_location = Text()
+    lifecycle_stage = Text()
+    tags = Nested(ElasticExperimentTag)
+
+    class Index:
+        name = 'mlflow-experiments'
+        settings = {
+            "number_of_shards": 1,
+            "number_of_replicas": 1
+        }
+
+    def to_mlflow_entity(self):
+        return Experiment(
+            experiment_id=str(self.meta.id),
+            name=self.name,
+            artifact_location=self.artifact_location,
+            lifecycle_stage=self.lifecycle_stage,
+            tags=[t.to_mlflow_entity() for t in self.tags])
+
+
+class ElasticMetric(InnerDoc):
+    key = Keyword()
+    value = Float()
+    timestamp = Text()
+    step = Integer()
+
+    def to_mlflow_entity(self):
+        return Metric(
+            key=self.key,
+            value=self.value,
+            timestamp=self.timestamp,
+            step=self.step)
+
+
+class ElasticLatestMetric(InnerDoc):
+    key = Keyword()
+    value = Float()
+    timestamp = Text()
+    step = Integer()
+
+
+class ElasticParam(InnerDoc):
+    key = Keyword()
+    value = Text()
+
+    def to_mlflow_entity(self):
+        return Param(
+            key=self.key,
+            value=self.value)
+
+
+class ElasticTag(InnerDoc):
+    key = Keyword()
+    value = Text()
+
+    def to_mlflow_entity(self):
+        return RunTag(
+            key=self.key,
+            value=self.value)
+
+
+class ElasticRun(Document):
+
+    name = Keyword()
+    source_type = Keyword()
+    source_name = Keyword()
+    experiment_id = Keyword()
+    user_id = Keyword()
+    status = Keyword()
+    start_time = Text()
+    end_time = Text()
```

Comment:
We need to be able to order by start time and end time, is there a time/datetime type in elasticsearch ?

Reply:
Yes it exists

Reply:
so I think we should use this type